codec-compression/src/main/java/io/netty/handler/codec/compression/ZstdConstants.java

```diff
@@ -37,7 +37,7 @@ final class ZstdConstants {
     /**
      * Max block size
      */
-    static final int MAX_BLOCK_SIZE = 1 << (DEFAULT_COMPRESSION_LEVEL + 7) + 0x0F;   //  32 M
+    static final int MAX_BLOCK_SIZE = Integer.MAX_VALUE;
```

Comment:
I have an issue interpreting the RFC.

https://datatracker.ietf.org/doc/html/rfc8478#section-3.1.1.2.3

I read:

>    The upper 21 bits of Block_Header represent the Block_Size.
   Block_Size is the size of the block excluding the header.  A block
   can contain any number of bytes (even zero), up to
   Block_Maximum_Decompressed_Size, which is the smallest of:
> -  Window_Size
> -  128 KB

Does it mean a blockSize has to be lower than 128KB ? It looks fairly low to me, and it may be unrelated to this PR change (I'm no expert in Zstd).

I mean, do we have a unitTest actually testing a large blockSize? (again blockSize != decodeSize, and I understand decodeSize (as worded in Netty) refers to windowSize (as referred in the RFC)).